var/spack/repos/builtin/packages/gnupg/package.py

```diff
@@ -43,6 +43,11 @@ class Gnupg(AutotoolsPackage):
     depends_on('iconv')
     depends_on('zlib')
 
+    @run_after('install')
+    @when('@2.0:2.999')
+    def add_gpg2_symlink(self):
+        symlink('gpg', join_path(self.prefix, 'bin', 'gpg2'))
```

Comment:
I didn't try this, but where is the symlink created? Shouldn't this be:
```suggestion
        with working_dir(self.prefix.bin):
            symlink('gpg', 'gpg2')
```
Or are these two implementations equivalent?

Reply:
It's equivalent, I don't see a reason to commit this change ^

Reply:
Could also use:
```suggestion
        symlink('gpg', self.prefix.bin.gpg2)
```

Reply:
Do we have other use cases outside of Spack where this is useful? If so I wouldn't mind doing the symlink in the package. Otherwise we might try first to add an override via an environment variable.

Reply:
Yes, lots of software first look for gpg2 then gpg, because gpg may be gpg 1.x, as is the case for ubuntu 16.04:
```

Reply:
ubuntu 18:04:

```
$ docker run --rm ubuntu:18.04 bash -c 'apt-get -yqq update &> /dev/null && apt-get -yqq install --no-install-recommends gpg &> /dev/null && gpg2 --version'
bash: gpg2: command not found
$ docker run --rm ubuntu:18.04 bash -c 'apt-get -yqq update &> /dev/null && apt-get -yqq install --no-install-recommends gpg &> /dev/null && gpg --version'
gpg (GnuPG) 2.2.4
```

Reply:
In fact the gpg2 symlink to gpg is in a different pacakge on ubuntu:

```
harmen-desktop ~ $ dpkg -S /usr/bin/gpg
gpg: /usr/bin/gpg
harmen-desktop ~ $ dpkg -S /usr/bin/gpg2
gnupg2: /usr/bin/gpg2
```

Reply:
So, ubuntu 16.04 is probably the last distribution where you can install both gpg 1.x and gpg 2.x next to eachother:

```
$ docker run --rm ubuntu:16.04 bash -c 'apt-get -yqq update &> /dev/null && apt-get -yqq install gnupg2 &> /dev/null && gpg2 --version && gpg --version'
gpg (GnuPG) 2.1.11
gpg (GnuPG) 1.4.20
```